be/src/exec/pipeline/scan/scan_operator.h

```diff
@@ -69,6 +82,8 @@ class ScanOperator : public SourceOperator {
     std::atomic<int> _num_running_io_tasks = 0;
     std::vector<std::atomic<bool>> _is_io_task_running;
     std::vector<ChunkSourcePtr> _chunk_sources;
+    mutable SpinLock _scan_status_mutex;
+    Status _scan_status;
```

Comment:
why need a special scan status, if storage engine return error status, we could pass the status to pipeline driver.

Reply:
> why need a special scan status, if storage engine return error status, we could pass the status to pipeline driver.

this error is returned by chunk source in io threads which can not be get directly by pipeline driver, we should record it first